src/common/proxy_wasm/ngx_proxy_wasm_properties.c

```diff
@@ -877,25 +879,58 @@ ngx_proxy_wasm_properties_get_host(ngx_proxy_wasm_ctx_t *pwctx,
 
     hpn = (host_props_node_t *)
               ngx_str_rbtree_lookup(&pwctx->host_props_tree, path, hash);
+    if (!hpn) {
+        return NGX_DECLINED;
+    }
 
-    if (hpn) {
-        value->data = hpn->value.data;
-        value->len = hpn->value.len;
+    /* stored value is not const: need to call getter again */
+    if (!hpn->is_const && pwctx->host_prop_getter) {
+        return NGX_DECLINED;
+    }
 
-        return NGX_OK;
+    value->data = hpn->value.data;
+    value->len = hpn->value.len;
+
+    if (hpn->negative_cache) {
```

Comment:
Should this be hoisted up above the previous 2 lines or do they have to run in this case? I keep forgetting what the negative cache is about, maybe some comments would help.

Reply:
It does seem to be for consts only; maybe we only set this bit when `is_const` is enabled? This way it'd be easier to trace-out what this bit tries to relate?

Reply:
I renamed `negative_cache` to `const_null`

Reply:
It seems like many changes from the early reviews are gone?